skrub/datasets/tests/test_utils.py

```diff
@@ -1,18 +1,54 @@
+import shutil
+import tempfile
 from pathlib import Path
-from unittest import mock
 
+from skrub.datasets._utils import get_data_dir, get_data_home
 
-@mock.patch("os.path.dirname")
-def test_get_data_dir(mock_os_path_dirname):
+
+def test_get_data_dir():
     """
     Tests function ``get_data_dir()``.
     """
-    from skrub.datasets._utils import get_data_dir
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmpdirpath = Path(tmpdirname)
+        assert get_data_dir(data_home=tmpdirpath) == tmpdirpath
+
+        assert get_data_dir(name="tests", data_home=tmpdirpath) == tmpdirpath / "tests"
+
+
+def test_get_data_home_str():
+    """
+    Test function for ``get_data_home()`` when `data_home` is a string.
```

Comment:
It looks like you are passing a path already. It looks redundant with the previous test. You might just want to add the assert above and try for both `str` and path-like with the parametrize.